orderable/models.py

```diff
@@ -101,25 +80,53 @@ def _move_to_new_pos():
             #     sort_order > new_pos.
             to_shift = to_shift.filter(sort_order__gte=self.sort_order)
             to_shift.update(sort_order=models.F('sort_order') + 1)
-            _move_to_new_pos()
+            self.sort_order = new_pos
 
         # self.sort_order decreased.
         elif old_pos and new_pos < old_pos:
-            _move_to_end()
+            self._move_to_end(objects)
+            super(Orderable, self).save()
             # Increment `sort_order` on objects with:
             #     sort_order >= new_pos and sort_order < old_pos
             to_shift = to_shift.filter(sort_order__gte=new_pos, sort_order__lt=old_pos)
             to_shift.update(sort_order=models.F('sort_order') + 1)
-            _move_to_new_pos()
+            self.sort_order = new_pos
 
         # self.sort_order increased.
         elif old_pos and new_pos > old_pos:
-            _move_to_end()
+            self._move_to_end(objects)
+            super(Orderable, self).save()
             # Decrement sort_order on objects with:
             #     sort_order <= new_pos and sort_order > old_pos.
             to_shift = to_shift.filter(sort_order__lte=new_pos, sort_order__gt=old_pos)
             to_shift.update(sort_order=models.F('sort_order') - 1)
-            _move_to_new_pos()
+            self.sort_order = new_pos
+
+    def _move_to_end(self, objects):
+        """Temporarily save `self.sort_order` elsewhere (max_obj)."""
+        max_obj = objects.all().aggregate(models.Max('sort_order'))['sort_order__max']
+        self.sort_order = max_obj + 1 if max_obj else 1
+
+    def save(self, *args, **kwargs):
+        """
+        Keep the unique order in sync.
+
+        Expects to be run in a transaction to avoid race conditions.
+
+        WARNING: Intensive giggery-pokery zone.
+        """
+        objects = self.get_filtered_manager()
+        old_pos = getattr(self, '_original_sort_order', None)
+        new_pos = self.sort_order
+
+        try:
+            with transaction.atomic():
+                self._save(objects, old_pos, new_pos)
+        except IntegrityError:
+            with transaction.atomic():
+                old_pos = objects.filter(pk=self.pk).values_list(
+                    'sort_order', flat=True)[0]
```

Comment:
I get the impression that this is the crux of the PR.

Could you please add a comment here to explain how/why an `IntegrityError` may occur?